cmd/prober: TestExcludeInRunAll flake

https://github.com/tailscale/tailscale/actions/runs/22409233231/job/64877799762?pr=18805
```
=== RUN   TestExcludeInRunAll
    prober_test.go:840: unexpected JSON response (-want +got):
          prober.RunHandlerAllResponse{
          	Results: map[string]prober.RunHandlerResponse{
          		"includedProbe": {
          			ProbeInfo: prober.ProbeInfo{
          				... // 3 ignored and 4 identical fields
          				Status: "succeeded",
          				Error:  "",
          				RecentResults: []bool{
          					true,
        - 					true,
          				},
          				... // 1 ignored field
          			},
        - 			PreviousSuccessRatio:  1,
        + 			PreviousSuccessRatio:  0,
          			PreviousMedianLatency: s"0s",
          		},
          	},
          }
--- FAIL: TestExcludeInRunAll (0.00s)
FAIL
FAIL	tailscale.com/prober	3.282
```

--- a/prober/prober_test.go
+++ b/prober/prober_test.go
@@ -793,9 +793,14 @@ func TestExcludeInRunAll(t *testing.T) {
 		},
 	}
 
-	p.Run("includedProbe", probeInterval, nil, FuncProbe(func(context.Context) error { return nil }))
-	p.Run("excludedProbe", probeInterval, nil, FuncProbe(func(context.Context) error { return nil }))
-	p.Run("excludedOtherProbe", probeInterval, nil, FuncProbe(func(context.Context) error { return nil }))
+	includedProbe := p.Run("includedProbe", probeInterval, nil, FuncProbe(func(context.Context) error { return nil }))
+	excludedProbe := p.Run("excludedProbe", probeInterval, nil, FuncProbe(func(context.Context) error { return nil }))
+	excludedOtherProbe := p.Run("excludedOtherProbe", probeInterval, nil, FuncProbe(func(context.Context) error { return nil }))
+
+	// Wait for all probes to complete their initial run
+	<-includedProbe.stopped
+	<-excludedProbe.stopped
+	<-excludedOtherProbe.stopped
 
 	mux := http.NewServeMux()
 	server := httptest.NewServer(mux)

